MekHQ/src/mekhq/campaign/stratcon/StratconRulesManager.java

```diff
@@ -0,0 +1,1080 @@
+            } else {
+                backingScenario.addScenarioModifier(AtBScenarioModifier.getScenarioModifier(AtBScenarioModifier.SCENARIO_MODIFIER_ALLIED_GROUND_UNITS));
+            }
+        }
+    }
+
+    /**
+     * Set the 'attached' units modifier for the current scenario (integrated, house, liaison),
+     * and make sure we're not deploying ground units to an air scenario
+     * @param contract The scenario's contract
+     */
+    public static void setAttachedUnitsModifier(StratconScenario scenario, AtBContract contract) {
+        AtBDynamicScenario backingScenario = scenario.getBackingScenario();
+        boolean airBattle = (backingScenario.getTemplate().mapParameters.getMapLocation() == MapLocation.LowAtmosphere) ||
+                (backingScenario.getTemplate().mapParameters.getMapLocation() == MapLocation.Space);
+        
+        // if we're on cadre duty, we're getting three trainees, period
+        if(contract.getMissionType() == AtBContract.MT_CADREDUTY) {
+            if(airBattle) {
+                backingScenario.addScenarioModifier(AtBScenarioModifier.getScenarioModifier(AtBScenarioModifier.SCENARIO_MODIFIER_TRAINEES_AIR));                
+            } else {
+                backingScenario.addScenarioModifier(AtBScenarioModifier.getScenarioModifier(AtBScenarioModifier.SCENARIO_MODIFIER_TRAINEES_GROUND));
+            }
+            return;
+        }
+        
+        // if we're under non-independent command rights, a supervisor may come along
+        switch(contract.getCommandRights()) {
+        case AtBContract.COM_INTEGRATED:
+            if(airBattle) {
+                backingScenario.addScenarioModifier(AtBScenarioModifier.getScenarioModifier(AtBScenarioModifier.SCENARIO_MODIFIER_INTEGRATED_UNITS_AIR));                
+            } else {
+                backingScenario.addScenarioModifier(AtBScenarioModifier.getScenarioModifier(AtBScenarioModifier.SCENARIO_MODIFIER_INTEGRATED_UNITS_GROUND));
+            }
+            break;
+        case AtBContract.COM_HOUSE:
+            if(airBattle) {
+                backingScenario.addScenarioModifier(AtBScenarioModifier.getScenarioModifier(AtBScenarioModifier.SCENARIO_MODIFIER_HOUSE_CO_AIR));
+            } else {
+                backingScenario.addScenarioModifier(AtBScenarioModifier.getScenarioModifier(AtBScenarioModifier.SCENARIO_MODIFIER_HOUSE_CO_GROUND));
+            }            
+            break;
+        case AtBContract.COM_LIAISON:
+            if(scenario.isRequiredScenario()) {
+                if(airBattle) {
+                    backingScenario.addScenarioModifier(AtBScenarioModifier.getScenarioModifier(AtBScenarioModifier.SCENARIO_MODIFIER_LIAISON_AIR));
+                } else {
+                    backingScenario.addScenarioModifier(AtBScenarioModifier.getScenarioModifier(AtBScenarioModifier.SCENARIO_MODIFIER_LIAISON_GROUND));
+                } 
+            }
+            break;
+        }
```

Comment:
Usually I think ternaries impair readability, but in this case with how long these lines are, consider making these ternaries on `airBattle ? X : Y`